%YAML:1.0
cameras:
    # Valid Fall 2018 and Spring 2019 to present (Summer 2021), not valid for Barbados 2019
    - {T_SC:
       [-0.05832554, -0.49729442, 0.86561908, 0.4404166,
         0.99825455, -0.02099889, 0.05519876, -0.04791664,
         -0.009273, 0.86732768, 0.49765118, -0.02872042,
         0., 0., 0., 1.],
       image_dimension: [800, 600],
       distortion_coefficients: [0.0, 0.0, 0.0, 0.0],
       distortion_type: radialtangential,
       focal_length: [579.92076363, 579.92076363],
       principal_point: [326.80654526, 312.22106171]}  # alpha = 0.0

    - {T_SC:
       [-0.04471672, -0.47414334,  0.87931138,  0.43962862,
        0.9982831,  -0.0545409,   0.02135733,  0.04401349,
        0.037832,    0.87875672,  0.47576818, -0.02956853,
        0., 0., 0., 1.],
       image_dimension: [800, 600],
       distortion_coefficients: [0.0, 0.0, 0.0, 0.0],
       distortion_type: radialtangential,
       focal_length: [579.92076363, 579.92076363],
       principal_point: [326.80654526, 312.22106171]}  # alpha = 0.0

camera_params:
    camera_rate: 30
    sigma_absolute_translation:  0.0 #8.0e-3
    sigma_absolute_orientation: 0.0 #8.0e-3
    sigma_c_relative_translation: 0.0 #1.0e-6 # below 1.0e-6 ruines numerics...
    sigma_c_relative_orientation: 0.0 #1.0e-6 # below 1.0e-6 ruines numerics...


imu_params:
    a_max: 176.0 # acceleration saturation [m/s^2]
    g_max: 7.8 # gyro saturation [rad/s]
    sigma_g_c: 0.0016017 # gyro noise density [rad/s/sqrt(Hz)] 0.00016017
    sigma_a_c: 0.0071376 # accelerometer noise density [m/s^2/sqrt(Hz)] 0.00071376
    sigma_bg: 0.03 # gyro bias prior [rad/s]
    sigma_ba: 0.1 # accelerometer bias prior [m/s^2]
    sigma_gw_c: 0.0000165 # gyro drift noise density [rad/s^s/sqrt(Hz)] 0.00000165
    sigma_aw_c: 0.0002874 # accelerometer drift noise density [m/s^2/sqrt(Hz)] 0.00002874
    tau: 3600.0 # reversion time constant, currently not in use [s]
    g: 9.81007 # Earth's acceleration due to gravity [m/s^2]
    a0: [ 0.0, 0.0, 0.0 ] # Accelerometer bias [m/s^2]
    imu_rate: 100
    # tranform Body-Sensor (IMU)
    # Sharmin Consider IMU frame as Body Frame
    T_BS:
        [1.0, 0.0, 0.0, 0.0,
         0.0, -1.0, 0.0, 0.0,
         0.0, 0.0, -1.0, 0.0,
         0.0, 0.0, 0.0, 1.0]

#gyroscope_noise_density: 0.2e-3, gyroscope_random_walk: 2.8577e-006

# Estimator parameters
numKeyframes: 5 # number of keyframes in optimisation window
numImuFrames: 3 # number of frames linked by most recent nonlinear IMU error terms

# ceres optimization options
ceres_options:
    minIterations: 3   # minimum number of iterations always performed
    maxIterations: 10  # never do more than these, even if not converged
    timeLimit: 0.05   # [s] negative values will set the an unlimited time limit

# detection
detection_options:
    threshold: 20.0      # detection threshold. By default the uniformity radius in pixels
    octaves: 1           # number of octaves for detection. 0 means single-scale at highest resolution
    maxNoKeypoints: 400  # restrict to a maximum of this many keypoints per image (strongest ones)   #Sharmin

# delay of images [s]:
imageDelay: 0.0  # in case you are using a custom setup, you will have to calibrate this. 0 for the VISensor.

# display debug images?
displayImages: true  # displays debug video and keyframe matches. May be slow.
publishDebugImages: false  # publish debug image to topic ~debug_image_x

# use direct driver
useDriver: false

# some options for how and what to publish -- optional in ROS-free version
publishing_options:
    publish_rate: 100                  # rate at which odometry updates are published only works properly if imu_rate/publish_rate is an integer!!
    publishLandmarks: true             # select, if you want to publish landmarks at all
    landmarkQualityThreshold: 1.0e-2   # landmark with lower quality will not be published
    maximumLandmarkQuality: 0.05       # landmark with higher quality will be published with the maximum colour intensity
    maxPathLength: 20                  # maximum length of the published path
    publishImuPropagatedState: false    # Should the state that is propagated with IMU messages be published? Or just the optimized ones?
    # provide custom World frame Wc
    T_Wc_W:
        [1.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 1.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 1.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 1.0000]
    trackedBodyFrame: B                # B or S, the frame of reference that will be expressed relative to the selected worldFrame
    velocitiesFrame: Wc                # Wc, B or S,  the frames in which the velocities of the selected trackedBodyFrame will be expressed in

#Sharmin
isSonarUsed: false
isDepthUsed: false
isRelocalization: true
histogramMethod: "CLAHE"     #NONE, HISTOGRAM, CLAHE
claheClipLimit: 2.0
claheTilesGridSize: 4

resizeFactor: 1.0  # by default 1.0. set this value if you want to resize the original image by this factor. You DO NOT need to change camera/projection matrix, those will be updated accordingly. 

isResetable: true  # Adds callbacks to allow hard and soft resets of SVIn2
